### ISSUE TITLE
Use apollo-client without `apollo-link-http`

### ISSUE THREAD

**Issue by tchak** [bgentry/ember-apollo-client#148]
I am implementing my own link (not based on http). If I exclude `apollo-link-http` from packages (to save some space) the apollo service will crash because unable to load `apollo-link-http`. Would you accept a PR to have a `BaseApolloService` without `apollo-link-http` (and maybe without cache) from which `ApolloService` will inherit and from which users can inherit in advanced cases?

**Comment by bgentry** [comment]
Yes, the design you describe makes a lot of sense. Ideally we should support the most common use case out-of-the-box with little/no required config, but also make it possible to serve more advanced use cases like yours.

Regarding the name, this is somewhat unconventional but what do you think about `ApolloBaseService` instead of `BaseApolloService`? I like it because it keeps the `Apollo` prefix.

**Comment by viniciussbs** [comment]
If this base service is a mixin, you should take a look at #139.
